python/tskit/trees.py

```diff
@@ -2498,14 +2498,35 @@ def map_mutations(self, genotypes, alleles):
         :param tuple(str) alleles: The alleles for the specified ``genotypes``. Each
             positive value in the ``genotypes`` array is treated as an index into this
             list of alleles.
+        :param ancestral_state: A fixed ancestral state, specified either as a
+            non-negative integer less than the number of alleles, or a string which
+            must be one of the ``alleles`` provided above. If ``None`` (default) then
+            the most parsimonious ancestral state is chosen. Note that if the ancestral
```

Comment:
"one of the most parsimonious ancestral states"

Reply:
Good point. I wonder if people will misread "one of the most" as "almost but not quite the best parsimony", because colloquially we say "one of the most" to mean "among the top XXX of their class". How about "an ancestral state is chosen which provides the most parsimonious placement of mutations"?

Reply:
It's not getting across the arbitrariness though, you're making it sound like there's a unique placement of mutations that gives the best parsimony, when there's really a single best parsimony *score* and often lots of ways of achieving that score, of which we choose one arbitrarily.

Also, I notice that we haven't emphasised that if you choose the ancestral state you will get back a parsimonious assignment of mutations conditioned on that ancestral state, and this may not be fully parsimonious.

Reply:
> It's not getting across the arbitrariness though, 

I hoped that saying "an ancestral state" rather than "the ancestral state" made that point. But perhaps that's too subtle?

> Also, I notice that we haven't emphasised that if you choose the ancestral state you will get back a parsimonious assignment of mutations conditioned on that ancestral state, and this may not be fully parsimonious.

Yes, good point. Will add.

Reply:
I now say 

            If ``None`` (default) then
            an ancestral state is chosen arbitrarily from among those that provide
            the most parsimonious placement of mutations. Note that if the ancestral
            state is specified, the placement of mutations may not be as parsimonious
            as that which could be achieved by leaving the ancestral state unspecified;
            additionally it may lead to mutations being placed above the root node(s) of
            the tree (for example if all the samples have a genotype of 1 but the
            ancestral state is fixed to be 0).

Which I think covers it all?